app/scripts/metamask-controller.js

```diff
@@ -4616,7 +4627,11 @@ export default class MetamaskController extends EventEmitter {
     try {
       // Scan accounts until we find an empty one
       const chainId = this.#getGlobalChainId();
+      let entropySource = keyringId;
 
+      if (!entropySource) {
+        entropySource = this.keyringController.state.keyringsMetadata[0].id;
+      }
```

Comment:
IMO, we shouldn't use the `KeyringController` state here, we should rely on the `withKeyring` above, IIRC we also pass-in the `metadata` now.

So we could do:
```ts
...
     const keyringSelector = keyringId
        ? { id: keyringId }
        : { type: KeyringTypes.hd };
      const { accounts, entropySource } = await this.keyringController.withKeyring(
        keyringSelector,
        async ({ keyring, metadata }) => {
          const accounts = await keyring.getAccounts();
          return {
            accounts,
            entropySource: metadata.id,
          };
        },
      );
...
```

WDYT?